### DISCUSSION TITLE
Interfacial energy at Grand Potential model

### DISCUSSION THREAD

**Discussion by RossMech** [idaholab/moose#19507] [Q&A Modules: Phase field] [answered]
Hallo everyone,

I am trying to describe equilibrium precipitation morphology in both twinned and untwinned state. In this purpose I use grand-potential like model. The relevant interfacial properties are:

- **Matrix-precipitate** interfacial energy lies approx. at **0.4 aJ/nm2**,
- **coherent twin interfacial energy** is equal to **0.09 aJ/nm2**
- **interfacial width** should lie around *0.3 nm*

Because of difference in magnitude of the interfacial energies, it is seems to be unpossible to find the values for gradient multiplicator **kappa**, double-well height **omega** and correspondent coeffecients **gamma**.

The GrandPotentialAction gives negative value for the first gamma coefficient. More details under link:
[(https://mooseframework.inl.gov/source/materials/GrandPotentialInterface.html#AagesenGP2018)]

The matlab script, attached to the link seems to solve the problem, but with various gradient coefficients **kappa_ij**. But I can not understand, how to implement such kernels, which would take different values of gradient coefficient at different interfaces. 

The question is then:

- Is it possible to find such a set of parameters to describe following interfacial properties using grand-potential model with the same gradient constant?
- If it is not, what would you suggest how to implement the gradient prefactors from matlab scipt or how to modify the model?

Thank in advance for your suggestions!

**Comment by simopier** [comment, marked as answer]
Hi, 
 
I ran into a similar issue in 2017 with a system that did not have such a large difference between the interfacial energies. Let me gather my notes and I will come back to you with some suggestions and answers. 

In the meantime, let me know if you have any updates.

Best,
PC

**Reply by RossMech** [reply]
Hi,

thank you for the answer. I've tried to implement following features to expand possible interfacial energies:

- Different kappa values for different phase fields. The interfacial energy can be reached, but a different width for two phases occur at the interface. I use Lagrange multiplier, which is localized on interface. If the interface of one phase overlaps with bulk of another one, then model becomes unbounded in [0..1] domain.
- The coefficient interpolation, as in the following paper of Moelans et al. https://journals.aps.org/prb/abstract/10.1103/PhysRevB.78.024113. The problem is, that i can't get the solver converged into a physical meaningfull solution.

I have tried both to introduce some small value into denominator to avoid zero values. As an alternative, I used condition to track a zero denominator and substitude the value of parameter with some small one.

**Reply by simopier** [reply]
@RossMech, I apologize for the delay. Below is a description of the problem and two approaches that you can use to solve it. The description is general, but I apply the first method to your system at the end. Please make sure to double check all the math and derivations, as this has not be checked by anyone or anything except my best intentions. 

Note: The notations used here will correspond to those used in the Physical Review B 2008 paper from Moelans et al. and the Physical Review E paper 2018 paper from Aagesen et al.

Goals: When selecting the parameters of a grand potential model, one aims at defining the parameters m, ![\kappa_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Ckappa_%7Bij%7D), ![\gamma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bij%7D) to obtain the desired (or close to desired) values for the interfacial energy ![\sigma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Csigma_%7Bij%7D) and interfacial thickness ![l_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7Bij%7D).

Introduction: It is first important to state that from the Physical Review B 2008 paper from Moelans et al., ![\sigma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Csigma_%7Bij%7D) and ![l_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7Bij%7D) are given by

(1).  ![\sigma_{ij}=g(\gamma_{ij}) \sqrt{m \kappa_{ij}}
](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Csigma_%7Bij%7D%3Dg%28%5Cgamma_%7Bij%7D%29+%5Csqrt%7Bm+%5Ckappa_%7Bij%7D%7D%0A) for ![\gamma_{ij} \in [0.53,2.5]
](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bij%7D+%5Cin+%5B0.53%2C2.5%5D%0A)

(2).  ![l_{ij} = \sqrt{\frac{\kappa_{ij}}{m f_o^{inter}(\gamma_{ij})}}
](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7Bij%7D+%3D+%5Csqrt%7B%5Cfrac%7B%5Ckappa_%7Bij%7D%7D%7Bm+f_o%5E%7Binter%7D%28%5Cgamma_%7Bij%7D%29%7D%7D%0A)

with ![g(\gamma_{ij})](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+g%28%5Cgamma_%7Bij%7D%29) a dimensionless function, which can be approximated as

(3).  ![g(\gamma_{ij}) = \frac{4}{3} \sqrt{\frac{2 \gamma_{ij} -1}{4(2 \gamma_{ij} +1)}}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+g%28%5Cgamma_%7Bij%7D%29+%3D+%5Cfrac%7B4%7D%7B3%7D+%5Csqrt%7B%5Cfrac%7B2+%5Cgamma_%7Bij%7D+-1%7D%7B4%282+%5Cgamma_%7Bij%7D+%2B1%29%7D%7D)

for ![\gamma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bij%7D) values around 1.5 (tabulated values also exists), and 

(4).  ![f_o^{inter}(\gamma_{ij})=\frac{2 \gamma_{ij} -1}{4(2 \gamma_{ij} +1)}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+f_o%5E%7Binter%7D%28%5Cgamma_%7Bij%7D%29%3D%5Cfrac%7B2+%5Cgamma_%7Bij%7D+-1%7D%7B4%282+%5Cgamma_%7Bij%7D+%2B1%29%7D) (tabulated values also exists).

These equations can be reversed to obtain m and ![\kappa](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Ckappa)as functions of ![\sigma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Csigma_%7Bij%7D) and ![l_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7Bij%7D), as

(5).  ![\kappa_{ij} = \frac{\sigma_{ij} l_{ij} \sqrt{(f_o^{inter}(\gamma_{ij}))}}{g(\gamma_{ij})}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Ckappa_%7Bij%7D+%3D+%5Cfrac%7B%5Csigma_%7Bij%7D+l_%7Bij%7D+%5Csqrt%7B%28f_o%5E%7Binter%7D%28%5Cgamma_%7Bij%7D%29%29%7D%7D%7Bg%28%5Cgamma_%7Bij%7D%29%7D)

(6).  ![m = \frac{\sigma_{ij}}{l_{ij} g(\gamma_{ij}) \sqrt{f_o^{inter}(\gamma_{ij})}}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+m+%3D+%5Cfrac%7B%5Csigma_%7Bij%7D%7D%7Bl_%7Bij%7D+g%28%5Cgamma_%7Bij%7D%29+%5Csqrt%7Bf_o%5E%7Binter%7D%28%5Cgamma_%7Bij%7D%29%7D%7D)

Note that ![g(\gamma_{ij})](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+g%28%5Cgamma_%7Bij%7D%29) and ![f_o^{inter}(\gamma_{ij})](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+f_o%5E%7Binter%7D%28%5Cgamma_%7Bij%7D%29) are not being simplified using Eqs. (3,4) to keep these equations as general as possible. 

It is also possible, using Eq. (1) and Eq. (3) (Valid only for ![\gamma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bij%7D) values around 1.5) to define ![\gamma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bij%7D) as

(7).  ![\gamma_{ij}= \frac{  m \kappa + \frac{9}{16} \sigma_{ij}^2}{2 m \kappa - \frac{9}{2}\sigma_{ij}^2}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bij%7D%3D+%5Cfrac%7B++m+%5Ckappa+%2B+%5Cfrac%7B9%7D%7B16%7D+%5Csigma_%7Bij%7D%5E2%7D%7B2+m+%5Ckappa+-+%5Cfrac%7B9%7D%7B2%7D%5Csigma_%7Bij%7D%5E2%7D).


As you described above, there are mostly two different ways to define the parameters of a grand potential model. One in which all the different interfaces use the same ![\kappa](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Ckappa), another one where each interface ij has a corresponding ![\kappa_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Ckappa_%7Bij%7D) value. We will cover both approaches and their limitations below. 

## Approach 1: ![\kappa](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Ckappa) is a constant for every phase and every interface

1. The interfacial energies ![\sigma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Csigma_%7Bij%7D) are either provided by the literature or values are assumed by the user.
2. A specific interface i1j1 is selected by the user.
3. The interfacial length for this interface ![l_{i1j1}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7Bi1j1%7D) is fixed by the user.
4.  ![\gamma_{i1j1}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bi1j1%7D) is also fixed by the user (1.5 is a good first guess).
5. ![f_o^{inter}(\gamma_{i1j1})](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+f_o%5E%7Binter%7D%28%5Cgamma_%7Bi1j1%7D%29) is calculated using Eq. (4) or tabulated values.
6. ![g(\gamma_{i1j1})](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+g%28%5Cgamma_%7Bi1j1%7D%29) is calculated using Eq. (3) or tabulated values.
7. ![\kappa](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Ckappa) and m can be derived using Eqs. (5,6) with ![\sigma_{i1j1}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Csigma_%7Bi1j1%7D), ![l_{i1j1}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7Bi1j1%7D), and ![\gamma_{i1j1}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bi1j1%7D).
8. The remaining ![l_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7Bij%7D) and ![\gamma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bij%7D) can be derived for the remaining interfaces using Eqs. (2,7). Note that every interface is expected to have different interfacial length and ![\gamma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bij%7D) values.
9. ![l_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7Bij%7D) and ![\gamma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bij%7D) values can vary a lot for each interface depending on the system, sometimes making it impossible to use. It is important to check that all the values of ![l_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7Bij%7D) and ![\gamma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bij%7D) are useable. For example, if ![\gamma_{ij} \notin [0.53,2.5]](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bij%7D+%5Cnotin+%5B0.53%2C2.5%5D), then Eq. (1) is not valid and the parameterization is incorrect. If ![l_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7Bij%7D) and ![\gamma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bij%7D) values are not satisfactory, the user should go back to step 4, 3 or maybe even step 2 of this process until reasonable values are found (not guaranteed). 

This approach is the simplest, but cannot necessarily lead to useable parameters, depending on the system. In which case the second approach can be an option. It should be possible to write a python code that tests different values of parameters and follow this process to find reasonable parameters.


## Approach 2: ![\kappa_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Ckappa_%7Bij%7D) depends on the interface

1. The interfacial energies ![\sigma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Csigma_%7Bij%7D) are either provided by the literature or values are assumed by the user.
2. A specific interface i1j1 is selected by the user.
3. The interfacial length for this interface ![l_{i1j1}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7Bi1j1%7D)is fixed by the user.
4.  ![\gamma_{i1j1}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bi1j1%7D) is fixed by the user around 1.5. 
5. m can be derived using Eq. (6).
6. Now that m is fixed, one can:
    a. Fix ![\gamma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bij%7D) values within [0.53,2.5] and derive the corresponding ![\kappa_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Ckappa_%7Bij%7D) and ![l_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7Bij%7D) values, which would be unique to each interface. In that case, the user should make sure that the ![l_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7Bij%7D) values are reasonable.
    b. Or fix the ![l_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7Bij%7D) values and derive the ![\gamma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bij%7D) values and ![\kappa_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Ckappa_%7Bij%7D) values. In that case, the user should make sure that the ![\gamma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bij%7D) values obtained at the end are within a range that keep the equations valid. For example, if Eq. (5) is used to derive ![\kappa_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Ckappa_%7Bij%7D), then ![\gamma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bij%7D) should be within [0.53,2.5].
    c. Or fix ![\kappa_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Ckappa_%7Bij%7D) and derive ![\gamma_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7Bij%7D) and ![l_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7Bij%7D) (Same comment than for b).


When using this approach, it is important to check if the final values are reasonable and within a range in which the equations are valid. If not, then the process should restart at step 6, 4, 3, or 2. Another thing to keep in mind is that, to my knowledge, there are currently no kernel in MOOSE to use different ![\kappa_{ij}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Ckappa_%7Bij%7D) values for each interface. Using this approach would require the user to write their own kernel. 

## Application of approach 1 to your system

Now, let's apply the first approach to your system, with 1-2 the matrix-precipitate interface, and 2-2 the precipitate-precipitate interface. 
1. You provided ![\sigma_{12}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Csigma_%7B12%7D)=0.4 (matrix ) and ![\sigma_{22}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Csigma_%7B22%7D)=0.09.
2. Let's select the interface 12 (matrix-precipitate).
3. You provided ![l_{12}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7B12%7D)=0.3.
4. Let's define  ![\gamma_{i1j2}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7B12%7D)=1.5.
5. f_o^{inter}( ![\gamma_{12}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7B12%7D))=0.125.
6. g( ![\gamma_{12}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7B12%7D))=0.4714.
7. ![\kappa](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Ckappa)=0.09 and m=8.
8. ![l_{22}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7B22%7D)=0.3 and  ![\gamma_{22}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7B22%7D)=0.516.
9. Unfortunately,  ![\gamma_{22}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7B22%7D) is too small for the equations to be valid. we need to find another set of parameters. You will find that changing  ![\gamma_{12}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7B12%7D) in step 4  will not help, neither will ![l_{12}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7B12%7D) in step 3. However, doing this process again from step 2 by selecting interface 22 and defining ![l_{22}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7B22%7D)=0.3 and  ![\gamma_{22}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7B22%7D)=1.5 provides ![\kappa](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Ckappa)=0.02025, m=1.8, ![l_{12}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7B12%7D)=0.3, and  ![\gamma_{12}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7B12%7D)=1.125, which fall within acceptable values. 

Let me know if you have any questions or concerns. Again, do not hesitate to re-do all the derivations before using these values. 

I hope that helped! If it did, feel free to mark this answer as a solution.

**Reply by simopier** [reply]
PS: you might not see the equations if using the dark mode. Use the light mode to see this page.

**Reply by RossMech** [reply]
Hello @simopier,

Thank you for a very detailed answer. I have tried to implement both of the schemes.

I had the following problems:
1) For the case of the constant kappa parameter, the difference in the diffuse interface widths was significant.
2) For the case of the variable kappa parameter, it was problematic to get a physically meaningful solution, especially in the triple point problem. I have thought that it can lie on the denominator stabilization. But it was not only the case. Also, I have checked a condition number and tried to improve the model to decrease it. It also did not help.

Highly possible it lies in the not stable behavior of the kappa derivative in the Allen-Cahn equation, see
https://journals.aps.org/prb/abstract/10.1103/PhysRevB.78.024113

Finally, it is to mention that a new preprint is published regarding this discussion. I found it very useful. The link is below
https://www.researchgate.net/publication/357580272_Comparison_of_three_different_strategies_to_introduce_anisotropic_interface_properties_in_a_multi-phase-field_model

So I would choose to use the first algorithm (with constant kappa parameter). It should be more expensive regarding memory usage, but it seems to converge better to the solution, especially at triple points.

**Reply by simopier** [reply]
Hi @RossMech, 

Thank you for the update and for the link to the preprint.

Were the diffuse interface widths really different? 

In my answer:

>## Application of approach 1 to your system
>
>Now, let's apply the first approach to your system, with 1-2 the matrix-precipitate interface, and 2-2 the precipitate-precipitate interface. 
>1. You provided ![\sigma_{12}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Csigma_%7B12%7D)=0.4 (matrix ) and ![\sigma_{22}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Csigma_%7B22%7D)=0.09.
>2. Let's select the interface 12 (matrix-precipitate).
>3. You provided ![l_{12}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7B12%7D)=0.3.
>4. Let's define  ![\gamma_{i1j2}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7B12%7D)=1.5.
>5. f_o^{inter}( ![\gamma_{12}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7B12%7D))=0.125.
>6. g( ![\gamma_{12}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7B12%7D))=0.4714.
>7. ![\kappa](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Ckappa)=0.09 and m=8.
>8. ![l_{22}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7B22%7D)=0.3 and  ![\gamma_{22}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7B22%7D)=0.516.
>9. Unfortunately,  ![\gamma_{22}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7B22%7D) is too small for the equations to be valid. we need to find another set of parameters. You will find that changing  ![\gamma_{12}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7B12%7D) in step 4  will not help, neither will ![l_{12}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7B12%7D) in step 3. However, doing this process again from step 2 by selecting interface 22 and defining ![l_{22}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7B22%7D)=0.3 and  ![\gamma_{22}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7B22%7D)=1.5 provides ![\kappa](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Ckappa)=0.02025, m=1.8, ![l_{12}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+l_%7B12%7D)=0.3, and  ![\gamma_{12}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cgamma_%7B12%7D)=1.125, which fall within acceptable values. 

I obtained the same diffuse interfacial width (0.3) for both. Were my calculations wrong?

Best,
PC

**Reply by RossMech** [reply]
@simopier 
It seems, that the calculations were a little bit of. 

If one takes the sigma_12 = 0.09 and gamma_12 = 1.5 (for coherent twin energy), then at my calculations we have matching values for kappa = 0.02025 and m = 1.8. But I had used not equation (7) but the equation solver for interfacial energy from equation (1) and also plotted the interfacial energy as function of gamma at constant kappa and m. For gamma_12 = 1.5 solver did not converge and the plot was following

![image](https://user-images.githubusercontent.com/62218067/152789285-398dde22-7202-40c8-8ef5-670628fd8534.png)

Then I changed the gamma_12 = 0.53 and the plot had a following form

![image](https://user-images.githubusercontent.com/62218067/152789507-2b43ea77-f92e-4b5c-9dc0-e6edbed4fdb1.png)

A suitable set of parameters was found as

kappa = 0.0203
m = 30.9
gamma_12 = 0.53 (twin interface)
gamma_13 = 1.85 (precipitate-matrix interface)

The corresponding width was estimated as:
l_12 = 0.3 (twin interface)
l_13 = 0.0675 (precipitate-matrix interface)
